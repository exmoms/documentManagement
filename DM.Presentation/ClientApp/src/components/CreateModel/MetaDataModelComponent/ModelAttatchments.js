import React from "react";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Autocomplete from "@material-ui/lab/Autocomplete";
import Button from "@material-ui/core/Button";
import DeleteIcon from "@material-ui/icons/Close";
import AddIcon from "@material-ui/icons/Add";
import { useTranslation } from "react-i18next";
import { makeStyles } from "@material-ui/core";

function CheckboxesTags(props) {
  var value = props.selectedMetaDataModels[props.index];

  if (!value) {
    value = { metaDataModelId: "", metaDataModelName: "" };
  }

  const { t } = useTranslation();
  return (
    <div>
      <Autocomplete
        name={"metaDataModels" + props.index}
        id={"modelName" + props.index}
        value={value}
        onChange={props.onChange}
        options={props.value.MetaDataModels}
        getOptionLabel={(option) =>
          option.metaDataModelName !== window.undefined
            ? option.metaDataModelName
            : ""
        }
        renderInput={(params) => (
          <TextField
            error={
              (value.metaDataModelId === "" ||
                value.metaDataModelId === window.undefined) &&
              props.value.step_2_error === true
                ? true
                : false
            }
            helperText={
              (value.metaDataModelId === "" ||
                value.metaDataModelId === window.undefined) &&
              props.value.step_2_error === true
                ? t("fill_out_field")
                : ""
            }
            {...params}
            label={t("selectmodel")}
            variant="outlined"
            fullWidth
          />
        )}
      />
    </div>
  );
}

const useStyles = makeStyles((theme) => ({
  textAlign: {
    textAlign: "initial",
  },
}));

export default function ModelAttachments(props) {
  const handleChange = (event) => {
    if (event.target.name === "isCompound") {
      props.handelisCompound(event.target.checked);
    } else if (event.target.name === "isAggregated") {
      props.handelisAggregated(event.target.checked);
    } else {
      var index = event.target.dataset.id;
      let compoundModels = [...props.value.compoundModels];

      if (event.target.name === "isRequired" + { index }) {
        compoundModels[index].isRequired = event.target.checked;
        props.handelcompoundModels(compoundModels);
      } else if (event.target.name === "nameOfAttatchment" + { index }) {
        compoundModels[index].caption = event.target.value;
        props.handelcompoundModels(compoundModels);
      }
    }
  };

  const handelTextField = (index, event, value) => {
    var aggregateMetaDataModelsParts = [
      ...props.value.aggregateMetaDataModelsParts,
    ];

    aggregateMetaDataModelsParts[index].aggregateName = event.target.value;
    props.handelaggregateMetaDataModelsParts(aggregateMetaDataModelsParts);
  };

  const handelSelect = (index, event, value) => {
    if (value) {
      var aggregateMetaDataModelsParts = [
        ...props.value.aggregateMetaDataModelsParts,
      ];
      aggregateMetaDataModelsParts[index].childMetaDataModelId = value.id;
      props.handelaggregateMetaDataModelsParts(aggregateMetaDataModelsParts);

      // store selected model from autoComplete
      var selectedMetaDataModels = [...props.value.selectedMetaDataModels];

      selectedMetaDataModels[index].metaDataModelId = value.id;
      selectedMetaDataModels[index].metaDataModelName = value.metaDataModelName;
      props.handelselectedMetaDataModels(selectedMetaDataModels);
    }
  };

  // addCompound function Adds new compound file for Model-------------------------
  const addCompound = () => {
    var compoundModels = [...props.value.compoundModels];
    compoundModels.push({ isRequired: false, caption: "" });
    props.handelcompoundModels(compoundModels);
  };
  // End----------------------------------

  // RemoveCompound function remove specific compound file from Model-------------------------
  const RemoveCompound = (index) => {
    var compoundModels = [...props.value.compoundModels];
    compoundModels.splice(index, 1);
    props.handelcompoundModels(compoundModels);
  };
  // End----------------------------------

  // addAggergate function Adds new file-------------------------
  const addAggergate = () => {
    var aggregateMetaDataModelsParts = [
      ...props.value.aggregateMetaDataModelsParts,
    ];
    aggregateMetaDataModelsParts.push({
      childMetaDataModelId: "",
      aggregateName: "",
    });
    props.handelaggregateMetaDataModelsParts(aggregateMetaDataModelsParts);

    var selectedMetaDataModels = [...props.value.selectedMetaDataModels];
    selectedMetaDataModels.push({
      metaDataModelId: "",
      metaDataModelName: "",
    });
    props.handelselectedMetaDataModels(selectedMetaDataModels);
  };
  // End----------------------------------

  // RemoveAggergate function remove specific file-------------------------
  const RemoveAggergate = (index) => {
    var aggregateMetaDataModelsParts = [
      ...props.value.aggregateMetaDataModelsParts,
    ];
    aggregateMetaDataModelsParts.splice(index, 1);
    props.handelaggregateMetaDataModelsParts(aggregateMetaDataModelsParts);

    // remove selected model from autoComplete
    var selectedMetaDataModels = [...props.value.selectedMetaDataModels];
    selectedMetaDataModels.splice(index, 1);
    props.handelselectedMetaDataModels(selectedMetaDataModels);
  };
  // End----------------------------------
  const { t } = useTranslation();
  const classes = useStyles();
  return (
    <React.Fragment>
      <Typography variant="h6" gutterBottom align="left">
        {t("modelproperties")}
      </Typography>
      <Grid container spacing={3}>
        <Grid item xs={3} className={classes.textAlign}>
          <FormControlLabel
            control={
              <Checkbox
                className="isCompound"
                color="secondary"
                name="isCompound"
                value={props.value.isCompound}
                checked={props.value.isCompound === true}
                onChange={handleChange}
              />
            }
            label={t("iscompound")}
          />
        </Grid>
        <Grid item xs={9} />
        <Grid item xs={12}>
          {/* This div will be full from the hook "componentDidUpdate" when isCompound is checked */}
          {/* <div id="AttatchmentsFiles"></div> */}
          {props.value.compoundModels.map((item, index) => (
            <div className="marginleft" key={index}>
              <TextField
                error={
                  item.caption === "" && props.value.step_2_error === true
                    ? true
                    : false
                }
                helperText={
                  item.caption === "" && props.value.step_2_error === true
                    ? t("fill_out_field")
                    : ""
                }
                id={"NameOfAttatchment" + index}
                required
                name={"nameOfAttatchment" + { index }}
                value={item.caption}
                inputProps={{ "data-id": index }}
                onChange={handleChange}
                label={t("nameofattatchment")}
                // helperText="Last three digits on signature strip"
                fullWidth
              />
              <Grid item xs={6} md={3} className={classes.textAlign}>
                <FormControlLabel
                  control={
                    <Checkbox
                      id={"isRequired" + index}
                      color="secondary"
                      name={"isRequired" + { index }}
                      inputProps={{ "data-id": index }}
                      value={item.isRequired}
                      checked={item.isRequired === true}
                      onChange={handleChange}
                    />
                  }
                  label={t("isrequired")}
                />
              </Grid>
              <Grid item xs={6} md={9} />
              <Grid item xs={12} md={6} className={classes.textAlign}>
                <Button
                  id="removeCompound"
                  variant="contained"
                  color="secondary"
                  startIcon={<DeleteIcon />}
                  onClick={() => RemoveCompound(index)}
                >
                  {t("remove")}
                </Button>
              </Grid>
            </div>
          ))}
          {props.value.isCompound && (
            <Grid item xs={12} className={classes.textAlign}>
              <br />
              <Button
                id="addCompound"
                variant="contained"
                onClick={addCompound}
                startIcon={<AddIcon />}
              >
                {t("addnewattribute")}
              </Button>
            </Grid>
          )}
          {/* ---------------------------------- */}
        </Grid>

        <Grid item xs={3} className={classes.textAlign}>
          <FormControlLabel
            control={
              <Checkbox
                className="isAggregated"
                color="secondary"
                name="isAggregated"
                value={props.value.isAggregated}
                checked={props.value.isAggregated === true}
                onChange={handleChange}
              />
            }
            label={t("isaggregated")}
          />
        </Grid>
        <Grid item xs={9} />
        <Grid item xs={12}>
          {/* This div will be full from the hook "componentDidUpdate" when isCompound is checked */}
          {/* <React.Fragment> */}
          <Grid container spacing={3}>
            <Grid item xs={12}>
              {/*  Show list of attributes*/}
              {props.value.aggregateMetaDataModelsParts.map((item, index) => (
                <div key={index} className="marginleft">
                  {/* <React.Fragment> */}
                  <Typography variant="h6" gutterBottom align="left">
                    {t("aggregated")}
                  </Typography>
                  <Grid container spacing={3}>
                    <Grid item xs={12} sm={6}>
                      <TextField
                        error={
                          item.aggregateName === "" &&
                          props.value.step_2_error === true
                            ? true
                            : false
                        }
                        helperText={
                          item.aggregateName === "" &&
                          props.value.step_2_error === true
                            ? t("fill_out_field")
                            : ""
                        }
                        id={"dataId" + index}
                        required
                        inputProps={{ "data-id": index }}
                        // id="Name"
                        name={"Name" + { index }}
                        label={t("namelabel")}
                        fullWidth
                        autoComplete="fname"
                        value={item.aggregateName}
                        onChange={(event, value) =>
                          handelTextField(index, event, value)
                        }
                      />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                      <CheckboxesTags
                        item={item}
                        value={props.value}
                        index={index}
                        selectedMetaDataModels={
                          props.value.selectedMetaDataModels
                        }
                        aggregateMetaDataModelsParts={
                          props.value.aggregateMetaDataModelsParts
                        }
                        onChange={(event, value) =>
                          handelSelect(index, event, value)
                        }
                        RemoveAggergate={RemoveAggergate}
                      />
                    </Grid>
                    <Grid item xs={12} sm={6} className={classes.textAlign}>
                      {/* </React.Fragment> */}
                      <Button
                        id="RemoveAggergate"
                        variant="contained"
                        color="secondary"
                        startIcon={<DeleteIcon />}
                        onClick={() => RemoveAggergate(index)}
                      >
                        {t("remove")}
                      </Button>
                    </Grid>
                  </Grid>
                </div>
              ))}
              {/* ----------------------------------- */}
            </Grid>

            {props.value.isAggregated && (
              <Grid item xs={12} className={classes.textAlign}>
                <Button
                  id="AddAggergate"
                  variant="contained"
                  onClick={addAggergate}
                  startIcon={<AddIcon />}
                >
                  {t("addnewfile")}
                </Button>
              </Grid>
            )}
          </Grid>
          {/* </React.Fragment> */}
          {/* ---------------------------------- */}
        </Grid>
      </Grid>
    </React.Fragment>
  );
}
