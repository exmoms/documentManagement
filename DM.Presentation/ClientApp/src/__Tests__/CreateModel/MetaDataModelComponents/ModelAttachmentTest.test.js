import React from "react";
import ReactDom from "react-dom";
import ModelAttatchments from "../../../components/CreateModel/MetaDataModelComponent/ModelAttatchments.js";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { act } from "react-dom/test-utils";
import ReactDOM from "react-dom";
afterEach(cleanup);
describe("ModelAttatchments", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = false;
    let isAggregatedMocked = false;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let MetaDataModelsMocked = [];
    let CompoundModelsMocked = [{ IsRequired: false, Caption: "" }];
    let SelectedMetaDataModelsMocked = [];
    ReactDom.render(
      <ModelAttatchments
        handelIsCompound={handelIsCompoundMock}
        handelIsAggregated={handelIsAggregatedMock}
        handelCompoundModels={handelCompoundModelsMock}
        handelAggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelSelectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />,
      div
    );
  });
  test("matches snapshot", async () => {
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = true;
    let isAggregatedMocked = false;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let MetaDataModelsMocked = [];
    let CompoundModelsMocked = [{ IsRequired: false, Caption: "" }];
    let SelectedMetaDataModelsMocked = [];
    const wrapper = shallow(
      <ModelAttatchments
        handelIsCompound={handelIsCompoundMock}
        handelIsAggregated={handelIsAggregatedMock}
        handelCompoundModels={handelCompoundModelsMock}
        handelAggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelSelectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("handles props component", async () => {
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = true;
    let isAggregatedMocked = true;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let MetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
      { metaDataModelId: 2, metaDataModelName: "meta data model 2" },
    ];
    let CompoundModelsMocked = [
      { isRequired: false, caption: "tested value 1" },
      { isRequired: true, caption: "tested value 2" },
    ];
    let SelectedMetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
    ];
    let wrapper = mount(
      <ModelAttatchments
        handelisCompound={handelIsCompoundMock}
        handelisAggregated={handelIsAggregatedMock}
        handelcompoundModels={handelCompoundModelsMock}
        handelaggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelselectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />,
      { attachTo: document.body }
    );

    expect(wrapper.find(".isCompound").at(0).props().value).toBeTruthy();
    expect(wrapper.find(".isAggregated").at(0).props().value).toBeTruthy();
    //  //test CompoundModels setting
    act(() => {
      wrapper.useEffect;
    });
    wrapper.update();
    act(() => {
      wrapper
        .find(".isCompound")
        .at(0)
        .props()
        .onChange({ target: { name: "isCompound", checked: true } });
      wrapper
        .find(".isAggregated")
        .at(0)
        .props()
        .onChange({ target: { name: "isAggregated", checked: true } });
    });
    wrapper.update();
    expect(handelIsCompoundMock).toHaveBeenCalledTimes(1);
    expect(handelIsAggregatedMock).toHaveBeenCalledTimes(1);
    CompoundModelsMocked.map((item, index) => {
      expect(document.getElementById("NameOfAttatchment" + index).value).toBe(
        CompoundModelsMocked[index].caption
      );
      if (index == 0) {
        expect(document.getElementById("isRequired" + index).value).toBe(
          "false"
        );
      } else {
        expect(document.getElementById("isRequired" + index).value).toBe(
          "true"
        );
      }
    });

    //test AggregateMetaDataModelsParts setting
    AggregateMetaDataModelsPartsMocked.map((item, index) => {
      expect(document.getElementById("dataId" + index).value).toBe(
        AggregateMetaDataModelsPartsMocked[index].AggregateName
      );
      expect(document.getElementById("modelName" + index)).toBeInTheDocument();
      expect(document.getElementById("modelName" + index).value).toBe(
        SelectedMetaDataModelsMocked[index].metaDataModelName
      );
    });
  });

  test("test remove Field functionality for CompoundModels", async () => {
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = true;
    let isAggregatedMocked = true;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let MetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
      { metaDataModelId: 2, metaDataModelName: "meta data model 2" },
    ];
    let CompoundModelsMocked = [
      { IsRequired: false, Caption: "tested value 1" },
      { IsRequired: true, Caption: "tested value 2" },
    ];
    let SelectedMetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
    ];
    let wrapper = mount(
      <ModelAttatchments
        handelisCompound={handelIsCompoundMock}
        handelisAggregated={handelIsAggregatedMock}
        handelcompoundModels={handelCompoundModelsMock}
        handelaggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelselectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />,
      { attachTo: document.body }
    );
    act(() => {
      fireEvent.click(document.getElementById("removeCompound"));
    });
    wrapper.update();
    expect(handelCompoundModelsMock).toHaveBeenCalledTimes(1);
    let CompoundModelsArgument = CompoundModelsMocked;
    CompoundModelsArgument.splice(0, 1);
    expect(handelCompoundModelsMock.mock.calls[0][0]).toStrictEqual(
      CompoundModelsArgument
    );
  });

  test("test add Field functionality for CompoundModels", async () => {
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = true;
    let isAggregatedMocked = true;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let MetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
      { metaDataModelId: 2, metaDataModelName: "meta data model 2" },
    ];
    let CompoundModelsMocked = [
      { IsRequired: false, Caption: "tested value 1" },
      { IsRequired: true, Caption: "tested value 2" },
    ];
    let SelectedMetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
    ];
    let wrapper = mount(
      <ModelAttatchments
        handelisCompound={handelIsCompoundMock}
        handelisAggregated={handelIsAggregatedMock}
        handelcompoundModels={handelCompoundModelsMock}
        handelaggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelselectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />,
      { attachTo: document.body }
    );
    act(() => {
      fireEvent.click(document.getElementById("addCompound"));
    });
    wrapper.update();
    expect(handelCompoundModelsMock).toHaveBeenCalledTimes(1);
    let CompoundModelsArgument = CompoundModelsMocked;
    CompoundModelsArgument.push({ isRequired: false, caption: "" });
    expect(handelCompoundModelsMock.mock.calls[0][0]).toStrictEqual(
      CompoundModelsArgument
    );
  });

  test("test remove Field functionality for AggregateMetaDataModelsParts", async () => {
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = true;
    let isAggregatedMocked = true;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", AggregateName: "" },
    ];
    let MetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
      { metaDataModelId: 2, metaDataModelName: "meta data model 2" },
    ];
    let CompoundModelsMocked = [
      { IsRequired: false, Caption: "tested value 1" },
      { IsRequired: true, Caption: "tested value 2" },
    ];
    let SelectedMetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
    ];
    let wrapper = mount(
      <ModelAttatchments
        handelisCompound={handelIsCompoundMock}
        handelisAggregated={handelIsAggregatedMock}
        handelcompoundModels={handelCompoundModelsMock}
        handelaggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelselectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />,
      { attachTo: document.body }
    );
    act(() => {
      fireEvent.click(document.getElementById("RemoveAggergate"));
    });
    wrapper.update();
    expect(handelAggregateMetaDataModelsPartsMock).toHaveBeenCalledTimes(1);
    let AggregateMetaDataModelsPartsArgument = AggregateMetaDataModelsPartsMocked;
    AggregateMetaDataModelsPartsArgument.splice(0, 1);
    expect(
      handelAggregateMetaDataModelsPartsMock.mock.calls[0][0]
    ).toStrictEqual(AggregateMetaDataModelsPartsArgument);
  });

  test("test add Field functionality for AggregateMetaDataModelsParts", async () => {
    const handelIsCompoundMock = jest.fn();
    const handelIsAggregatedMock = jest.fn();
    const handelCompoundModelsMock = jest.fn();
    const handelAggregateMetaDataModelsPartsMock = jest.fn();
    const handelSelectedMetaDataModelsMock = jest.fn();

    let isCompoundMocked = true;
    let isAggregatedMocked = true;
    let AggregateMetaDataModelsPartsMocked = [
      { childMetaDataModelId: "", aggregateName: "" },
    ];
    let MetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
      { metaDataModelId: 2, metaDataModelName: "meta data model 2" },
    ];
    let CompoundModelsMocked = [
      { IsRequired: false, Caption: "tested value 1" },
      { IsRequired: true, Caption: "tested value 2" },
    ];
    let SelectedMetaDataModelsMocked = [
      { metaDataModelId: 1, metaDataModelName: "meta data model 1" },
    ];
    let wrapper = mount(
      <ModelAttatchments
        handelisCompound={handelIsCompoundMock}
        handelisAggregated={handelIsAggregatedMock}
        handelcompoundModels={handelCompoundModelsMock}
        handelaggregateMetaDataModelsParts={
          handelAggregateMetaDataModelsPartsMock
        }
        handelselectedMetaDataModels={handelSelectedMetaDataModelsMock}
        value={{
          isCompound: isCompoundMocked,
          isAggregated: isAggregatedMocked,
          aggregateMetaDataModelsParts: AggregateMetaDataModelsPartsMocked,
          MetaDataModels: MetaDataModelsMocked,
          compoundModels: CompoundModelsMocked,
          selectedMetaDataModels: SelectedMetaDataModelsMocked,
        }}
      />,
      { attachTo: document.body }
    );
    act(() => {
      fireEvent.click(document.getElementById("AddAggergate"));
    });
    wrapper.update();
    expect(handelAggregateMetaDataModelsPartsMock).toHaveBeenCalledTimes(1);
    let AggregateMetaDataModelsPartsArgument = AggregateMetaDataModelsPartsMocked;
    AggregateMetaDataModelsPartsArgument.push({
      childMetaDataModelId: "",
      aggregateName: "",
    });
    expect(
      handelAggregateMetaDataModelsPartsMock.mock.calls[0][0]
    ).toStrictEqual(AggregateMetaDataModelsPartsArgument);
  });
});
