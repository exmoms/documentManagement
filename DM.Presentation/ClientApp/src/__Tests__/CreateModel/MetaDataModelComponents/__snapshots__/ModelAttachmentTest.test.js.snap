// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelAttatchments matches snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    align="left"
    gutterBottom={true}
    variant="h6"
  >
    modelproperties
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-textAlign-1"
      item={true}
      xs={3}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        control={
          <ForwardRef(WithStyles)
            checked={true}
            className="isCompound"
            color="secondary"
            name="isCompound"
            onChange={[Function]}
            value={true}
          />
        }
        label="iscompound"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={9}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <div
        className="marginleft"
        key="0"
      >
        <WithStyles(ForwardRef(TextField))
          error={false}
          fullWidth={true}
          helperText=""
          id="NameOfAttatchment0"
          inputProps={
            Object {
              "data-id": 0,
            }
          }
          label="nameofattatchment"
          name="nameOfAttatchment[object Object]"
          onChange={[Function]}
          required={true}
        />
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-textAlign-1"
          item={true}
          md={3}
          xs={6}
        >
          <WithStyles(ForwardRef(FormControlLabel))
            control={
              <ForwardRef(WithStyles)
                checked={false}
                color="secondary"
                id="isRequired0"
                inputProps={
                  Object {
                    "data-id": 0,
                  }
                }
                name="isRequired[object Object]"
                onChange={[Function]}
              />
            }
            label="isrequired"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={9}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-textAlign-1"
          item={true}
          md={6}
          xs={12}
        >
          <WithStyles(ForwardRef(Button))
            color="secondary"
            id="removeCompound"
            onClick={[Function]}
            startIcon={<UNDEFINED />}
            variant="contained"
          >
            remove
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </div>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-textAlign-1"
        item={true}
        xs={12}
      >
        <br />
        <WithStyles(ForwardRef(Button))
          id="addCompound"
          onClick={[Function]}
          startIcon={<UNDEFINED />}
          variant="contained"
        >
          addnewattribute
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-textAlign-1"
      item={true}
      xs={3}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        control={
          <ForwardRef(WithStyles)
            checked={false}
            className="isAggregated"
            color="secondary"
            name="isAggregated"
            onChange={[Function]}
            value={false}
          />
        }
        label="isaggregated"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={9}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <div
            className="marginleft"
            key="0"
          >
            <WithStyles(ForwardRef(Typography))
              align="left"
              gutterBottom={true}
              variant="h6"
            >
              aggregated
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={3}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                sm={6}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="fname"
                  error={false}
                  fullWidth={true}
                  helperText=""
                  id="dataId0"
                  inputProps={
                    Object {
                      "data-id": 0,
                    }
                  }
                  label="namelabel"
                  name="Name[object Object]"
                  onChange={[Function]}
                  required={true}
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                sm={6}
                xs={12}
              >
                <CheckboxesTags
                  RemoveAggergate={[Function]}
                  aggregateMetaDataModelsParts={
                    Array [
                      Object {
                        "AggregateName": "",
                        "childMetaDataModelId": "",
                      },
                    ]
                  }
                  index={0}
                  item={
                    Object {
                      "AggregateName": "",
                      "childMetaDataModelId": "",
                    }
                  }
                  onChange={[Function]}
                  selectedMetaDataModels={Array []}
                  value={
                    Object {
                      "MetaDataModels": Array [],
                      "aggregateMetaDataModelsParts": Array [
                        Object {
                          "AggregateName": "",
                          "childMetaDataModelId": "",
                        },
                      ],
                      "compoundModels": Array [
                        Object {
                          "Caption": "",
                          "IsRequired": false,
                        },
                      ],
                      "isAggregated": false,
                      "isCompound": true,
                      "selectedMetaDataModels": Array [],
                    }
                  }
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                className="makeStyles-textAlign-1"
                item={true}
                sm={6}
                xs={12}
              >
                <WithStyles(ForwardRef(Button))
                  color="secondary"
                  id="RemoveAggergate"
                  onClick={[Function]}
                  startIcon={<UNDEFINED />}
                  variant="contained"
                >
                  remove
                </WithStyles(ForwardRef(Button))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
